workflows/ncn/storage/storage.upgrade.yaml

```diff
@@ -240,6 +252,62 @@ spec:
               {{ if ne $index 0 }}
               - check-ceph-health-{{ index $.TargetNcns (add $index -1) }}
               {{ end }}
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+                - name: scriptContent
+                  value: |
+                    # set maintenance mode on target_ncn
+                    TARGET_NCN={{$value}}
+                    if [[ -n $(ceph health detail | grep "$TARGET_NCN is in maintenance") ]]; then
+                      echo "$TARGET_NCN is already in maintenance mode"
+                    else
+                      echo "Setting ceph maintenance mode on $TARGET_NCN"
+                      ceph orch host maintenance enter $TARGET_NCN --force
+                      if [[ $? -ne 0 ]]; then
+                        echo "ERROR setting maintenance mode on $TARGET_NCN"
+                        exit 1
+                      fi
+                    fi
+                    # remove local images on target_ncn
+                    echo "Removing local images on $TARGET_NCN"
+                    ssh $TARGET_NCN podman rmi --all -f
+                    #exit maintenance mode on target_ncn
```

Comment:
we want to pull the ceph image prior to exiting maintenance mode.  the monitoring stack can be addressed like below, but we should do a pull and confirm the sha since it will have changed from when we pushed it.